creator-node/src/routes/files.js

```diff
@@ -484,6 +484,31 @@ module.exports = function (app) {
    */
   app.get('/ipfs/:dirCID/:filename', getDirCID)
 
+  app.post('/batch_cids_exist', handleResponse(async (req, res) => {
+    const { cids } = req.body
+
+    const queryResults = (await models.File.findAll({
+      attributes: ['multihash', 'storagePath'],
+      where: {
+        multihash: {
+          [models.Sequelize.Op.in]: cids
+        }
+      }
+    }))
+
+    const cidExists = {}
+    queryResults.forEach(({ multihash, storagePath }) => {
+      cidExists[multihash] = fs.existsSync(storagePath)
+    })
```

Comment:
this is pretty non-performant - it will do each sequentially with each existsSync call blocking eventloop

instead we should do something more parallelized like `await Promise.all(queryResults.map(...`
this is dangerous to do without a max concurrency though, so we'll prob need that as well


also instead of `fs.existsSync()` lets use the async pathExists from fs-extra package ([docs here](https://github.com/jprichardson/node-fs-extra/blob/HEAD/docs/pathExists.md)). also cmd+f for "pathExists" in creator-node to see other uses of this call